File: library/src/main/java/com/schibsted/spain/barista/BaristaEditTextActions.java
package com.schibsted.spain.barista;

import android.support.annotation.IdRes;
import android.support.test.espresso.PerformException;
import android.util.Log;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.action.ViewActions.replaceText;
import static com.schibsted.spain.barista.BaristaScrollActions.scrollTo;
import static com.schibsted.spain.barista.custom.DisplayedMatchers.displayedWithId;

public class BaristaEditTextActions {

  public static void writeToEditText(@IdRes int id, String text) {
    try {
      scrollTo(id);
    } catch (PerformException exception) {
      Log.d("Barista",
          "The View's parent is not a ScrollView. Due to the power of Barista, you can ignore this error message");
    } finally {
      onView(displayedWithId(id)).perform(replaceText(text));
    }
  }
}

File: library/src/main/java/com/schibsted/spain/barista/BaristaEditTextActions.kt
package com.schibsted.spain.barista

import android.support.annotation.IdRes
import android.support.test.espresso.Espresso.onView
import android.support.test.espresso.action.ViewActions.replaceText
import android.support.test.espresso.matcher.ViewMatchers.withId
import com.schibsted.spain.barista.BaristaScrollActions.safelyScrollTo
import com.schibsted.spain.barista.custom.DisplayedMatchers.displayedAnd

object BaristaEditTextActions {

    @JvmStatic
    fun writeToEditText(@IdRes id: Int, text: String) {
        safelyScrollTo(id)
        onView(displayedAnd(withId(id))).perform(replaceText(text))
    }
}
